test/test_fsdp2/test_fsdp2_eager.py

```diff
@@ -0,0 +1,448 @@
+import copy
+import threading
+import unittest
+from typing import Any, List
+
+import torch
+import torch._dynamo.testing
+import torch.distributed as dist
+import torch.nn as nn
+from float8_experimental.float8_dynamic_linear import (
+    Float8DynamicLinear,
+    Float8DynamicLinearWeightTensor,
+)
+from float8_experimental.float8_linear_utils import swap_linear_with_float8_linear
+from test_fsdp2_common import (
+    check_parity_bf16_mp,
+    check_parity_no_mp,
+    set_enable_fsdp_fp8_all_gather,
+)
+from torch.distributed._composable.fsdp import fully_shard, MixedPrecisionPolicy
+from torch.distributed._tensor import DTensor
+from torch.testing._internal.common_cuda import TEST_CUDA
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_fsdp import (
+    FSDPTest,
+    FSDPTestMultiThread,
+    MLP,
+    patch_all_gather,
+)
+from torch.testing._internal.common_utils import run_tests
+from torch.testing._internal.distributed._tensor.common_dtensor import (
+    ModelArgs,
+    Transformer,
+    TransformerBlock,
+)
+
+
+class TestFloat8Common:
```

Comment:
Another awesome follow up would be to use the distributed testing infra instead of the shellscripts we have for other distributed tests